### PR TITLE
Improve monitor efficiency by caching staker activity partition results for 2 hours at a time

### PR DESCRIPTION
**Pull request by derekpierre**
Related to #99 , #100 .

### CODE REVIEW

File: monitor/crawler.py [hunk 1]
```diff
@@ -328,17 +332,30 @@ def _measure_top_stakers(self) -> dict:
...
+                    # no active stake, ignore
+                    continue
+                inactive_without_expired.append(staker)
+
+            # update cache
+            self.__cached_staker_activity = confirmed, pending, inactive_without_expired
+            self.__cached_staker_activity_last_updated = maya.now()
+
+        # get data from cache
+        confirmed, pending, inactive = self.__cached_staker_activity
```
Comment (vzotova) [inline monitor/crawler.py:353]: What do you think about caching only expired and exclude them from `partition_stakers_by_activity`, in that case we will have more actual information but still not so much calls

Comment (derekpierre) [reply monitor/crawler.py:353]: Interesting, but I can't specifically exclude expired from the `partition_stakers_by_activity()` call without modifying the nucypher library - it current returns all three.

Unless you mean when using cached inactive stakers, to instead call `get_all_active_stakers()` for the current period, then separate those into active / pending. Then, when the cache is expired use  `partition_stakers_by_activity()` again...?

Comment (vzotova) [reply monitor/crawler.py:353]: missed that `partition_stakers_by_activity` is part of main lib, anyway I'm for modifying it if it's not to much

**Comment by derekpierre** [comment]
Perhaps for the moment, caching may be overkill. The scraping rounds on the status monitor that is running code from #100 (i.e. before this change) are a little over 1 minute, which is good enough. Having our own web3 node really helps:
```
crawler_1   | Scraping round completed (duration 0:01:57).
crawler_1   | Scraping round completed (duration 0:01:08).
crawler_1   | Scraping round completed (duration 0:01:22).
crawler_1   | Scraping round completed (duration 0:01:22).
crawler_1   | Scraping round completed (duration 0:01:02).
crawler_1   | Scraping round completed (duration 0:01:24).
crawler_1   | Scraping round completed (duration 0:01:13).
crawler_1   | Scraping round completed (duration 0:01:22).
crawler_1   | Scraping round completed (duration 0:01:25).
crawler_1   | Scraping round completed (duration 0:01:21).
crawler_1   | Scraping round completed (duration 0:01:22).
crawler_1   | Scraping round completed (duration 0:01:27).
crawler_1   | Scraping round completed (duration 0:01:09).
crawler_1   | Scraping round completed (duration 0:01:20).
crawler_1   | Scraping round completed (duration 0:01:18).
crawler_1   | Scraping round completed (duration 0:00:55).
...
```

and of that checking staker activity (i.e. running partition):
```
crawler_1   | ✓ ... Staker Confirmation Status [62.0s]
crawler_1   | ✓ ... Staker Confirmation Status [46.0s]
crawler_1   | ✓ ... Staker Confirmation Status [61.0s]
crawler_1   | ✓ ... Staker Confirmation Status [61.0s]
crawler_1   | ✓ ... Staker Confirmation Status [41.0s]
crawler_1   | ✓ ... Staker Confirmation Status [62.0s]
crawler_1   | ✓ ... Staker Confirmation Status [52.0s]
crawler_1   | ✓ ... Staker Confirmation Status [61.0s]
crawler_1   | ✓ ... Staker Confirmation Status [63.0s]
crawler_1   | ✓ ... Staker Confirmation Status [59.0s]
crawler_1   | ✓ ... Staker Confirmation Status [60.0s]
crawler_1   | ✓ ... Staker Confirmation Status [64.0s]
crawler_1   | ✓ ... Staker Confirmation Status [46.0s]
crawler_1   | ✓ ... Staker Confirmation Status [60.0s]
crawler_1   | ✓ ... Staker Confirmation Status [58.0s]
crawler_1   | ✓ ... Staker Confirmation Status [39.0s]
...
```

I'm trying to limit changes made to agents when contracts are going to change anyway.

Will close for now, but will keep an eye out and revisit later if there are efficiency problems.

**Pull request closed without merging**